src/train.py

```diff
@@ -62,24 +61,53 @@ def train(args):
         TODO: Include functionality for saving, loading from save
     '''
     set_seed(args.seed)
-    dataset = get_dataset()
+    print("#### Getting Dataset ####")
+    supervised_train_dataset = get_dataset('labeled_train.csv')
+    unsupervised_train_dataset = get_dataset('unlabeled_train.csv')
+    val_dataset = get_dataset('val.csv')
     # init models and optimizers
     model = None
     optimizer = None
 
     for epoch in range(args.epochs):
-        dataloader = DataLoader(dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+        # Training model
+        supervised_dataloader = DataLoader(supervised_train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+        supervised_iter = iter(supervised_dataloader)
+
+        unsupervised_dataloader = DataLoader(unsupervised_train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+
+        bar = tqdm(unsupervised_dataloader)
+        bar.set_description("Epoch {} Training Model".format(epoch))
+
+        # step counter for a single epoch
+        # used to determine which training task to do
+        epoch_step = 0
 
-        pbar = tqdm(dataloader)
-        for i, data in enumerate(pbar):
-            pbar.set_description("Processing at epoch %d"%epoch)
+        # We are considering one pass through the unsupervised dataset as
+        # one epoch
+        for unsupervised_batch in bar:
```

Comment:
I don't think this a bad approach at all, but perhaps it is worth considering this alternative approach from the NMT repo: https://github.com/facebookresearch/UnsupervisedMT/blob/d5f2fc29246205abd8d62a4377c2bd4c01e086b8/NMT/main.py#L270

The idea is we define an epoch as a number of training iterations, and in each training iteration, we have a number of supervised steps, a number of autoencoder steps, a number cross model steps, and a number of discriminator steps.  Thoughts?

Reply:
We could do this. How do we define the number of each number of steps for each task? The current approach is nice because it closely follows the TTS-ASR paper, and it guarantees that in a single epoch, we go through the entire dataset at least once, regardless of batch size.

Reply:
Yeah that is a good point.  I think this approach works well for now, we can comeback to this later if we need to change things